app/services/organization_update_service.rb

```diff
@@ -42,17 +44,29 @@ def update_partner_flags(organization)
     private
 
     def valid?(organization, params)
-      return true if organization.partners.none?
-      fields_marked_for_disabling = FIELDS.select { |field| params[field] == false }
+      fields_marked_for_disabling = FIELDS.select { |field| params[field] == "false" }
+      
       # Here we do a check: if applying the params for disabling request types to all
       # partners would mean any one partner would have all its request types disabled,
-      # then we should not apply the params. As per:
+      # then we should not apply the params and return an error message. As per:
       # github.com/rubyforgood/human-essentials/issues/3264
-      organization.partners.none? do |partner|
-        all_fields_will_be_disabled?(partner, fields_marked_for_disabling)
+      invalid_partner_names = find_invalid_partners(organization, fields_marked_for_disabling)
+      
+      if invalid_partner_names.empty?
+        return true
+      else
+        organization.errors.add(:base, "Please update #{invalid_partner_names.join(", ")}, so that they could make requests if you completed this change, then try again. Thank you.")
+        return false
       end
     end
 
+    def find_invalid_partners(organization, fields_marked_for_disabling)
+      #finds any partners who's request types will all be disabled
+      organization.partners.select do |partner|
+        all_fields_will_be_disabled?(partner, fields_marked_for_disabling)
+      end.map(&:name)
```

Comment:
The method indicates it's returning partners, not names. It's a bit cleaner to return the partners here and map the names in the calling code.

Reply:
Heard!